playground/main.cpp

```diff
@@ -248,14 +248,15 @@ struct sum {
 
     int yield() { _state = process_state::await; return _sum; }
 
-    void close() { _state = process_state::yield; }
+    void close() { _state = process_state::closed; }
```

Comment:
Not sure what "closed" for a state is- close is called when a process is in an await state to signal that no more data is coming. In response to a close, a process can switch to a yield state to yield values, otherwise it is destructed.